sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs

```diff
@@ -454,6 +454,70 @@ public static ExtractKeyPhrasesResultCollection ExtractKeyPhrasesResultCollectio
+        /// <param name="error">Sets the <see cref="TextAnalyticsResult.Error"/> property.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.ExtractSummaryResult"/> for mocking purposes.</returns>
+        public static ExtractSummaryResult ExtractSummaryResult(string id, TextAnalyticsError error)
+        {
+            return new ExtractSummaryResult(id, error);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.ExtractSummaryResultCollection"/> for mocking purposes.
+        /// </summary>
+        /// <param name="list">Sets the collection of <see cref="TextAnalytics.ExtractSummaryResult"/>.</param>
+        /// <param name="statistics">Sets the <see cref="ExtractSummaryResultCollection.Statistics"/> property.</param>
+        /// <param name="modelVersion">Sets the <see cref="ExtractSummaryResultCollection.ModelVersion"/> property.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.ExtractSummaryResultCollection"/> for mocking purposes.</returns>
+        public static ExtractSummaryResultCollection ExtractSummaryResultCollection(IEnumerable<ExtractSummaryResult> list, TextDocumentBatchStatistics statistics, string modelVersion)
+        {
+            return new ExtractSummaryResultCollection(list.ToList(), statistics, modelVersion);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.SummarySentence"/> for mocking purposes.
+        /// </summary>
+        /// <param name="text">Sets the <see cref="SummarySentence.Text"/> property.</param>
+        /// <param name="rankScore">Sets the <see cref="SummarySentence.RankScore"/> property.</param>
+        /// <param name="offset">Sets the <see cref="SummarySentence.Offset"/> property.</param>
+        /// <param name="length">Sets the <see cref="SummarySentence.Length"/> property.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.SummarySentence"/> for mocking purposes.</returns>
+        public static SummarySentence SummarySentence(string text, double rankScore, int offset, int length)
+        {
+            return new SummarySentence(new ExtractedSummarySentence(text, rankScore, offset, length));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.SummarySentenceCollection"/> for mocking purposes.
+        /// </summary>
+        /// <param name="sentences">Sets the collection of <see cref="TextAnalytics.SummarySentence"/>.</param>
+        /// <param name="warnings">Sets the <see cref="CategorizedEntityCollection.Warnings"/> property.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.SummarySentenceCollection"/> for mocking purposes.</returns>
+        public static SummarySentenceCollection SummarySentenceCollection(IList<SummarySentence> sentences, IList<TextAnalyticsWarning> warnings = default)
```

Comment:
could we change both of these to `IEnumerable`?

Reply:
That's a good point. I haven't payed attention to which collections we were using. From my understanding, we should use `IList` when we care about the order, and `IEnumerable` when it doesn't matter (is that right?).

Reply:
That is my understanding too. Guidelines recommend https://azure.github.io/azure-sdk/dotnet_introduction.html#supporting-types so we use `IEnumerable` . Looking at TA we have a combination of both :| Also, we always make them `List` by calling `ToList` so not sure it matters.

Reply:
> Also, we always make them List by calling ToList so not sure it matters.

The conversion from `IEnumerable` to `List` will depend on the type of collection being used. Users will probably give us a list if they care about order, so we preserve the order when calling `ToList`.

The auto-generated model factory seems to use `IEnumerable` even when the corresponding property is a list. For example, in Metrics Advisor:

Model:
https://github.com/Azure/azure-sdk-for-net/blob/fe8e623c8ef31810a4712fb12fec4ecbc019859f/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AnomalyDetection/IncidentRootCause.cs#L26

Factory:
https://github.com/Azure/azure-sdk-for-net/blob/fe8e623c8ef31810a4712fb12fec4ecbc019859f/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Generated/MicrosoftAzureMetricsAdvisorRestAPIOpenAPIV2ModelFactory.cs#L73

Given these points, I think it makes sense to go with `IEnumerable` everywhere. I'll update the new methods.